packages/svelte/src/compiler/phases/2-analyze/validation.js

```diff
@@ -579,6 +579,13 @@ export const validation_runes_js = {
 			...context.state,
 			private_derived_state
 		});
+	},
+	NewExpression(node, context) {
+		const callee = node.callee;
+
+		if (callee.type === 'ClassExpression' && context.state.scope.function_depth !== 0) {
```

Comment:
Just to clarify: What exactly does `context.state.scope.function_depth !== 0` relate to here? In other words, in which cases would this warning _not_ fire?

Reply:
Module level `new class` should still work.

Reply:
Ah yeah makes sense